### PR TITLE
feat(eza): create plugin

### PR DESCRIPTION
**Pull request by pepoluan**
The `eza` plugin provides some configurable aliases to simplify usage of the [`eza`](https://github.com/eza-community/eza) tool.

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

* Creates 5 aliases that are very common to invoke when using the `ls` command, but implemented using `eza` instead
* Allows limited configuration for `eza` defaults through usage of `zstyle` (see `README.md` for info)

### CODE REVIEW

File: plugins/eza/eza.plugin.zsh
Comment (carlosala) [inline plugins/eza/eza.plugin.zsh:25]: I'd instead use the same way of `zstyle` as in https://github.com/ohmyzsh/ohmyzsh/blob/c891b7e6ada5427f4c8cca05c8ed1ae33cb49943/plugins/yarn/yarn.plugin.zsh#L1-L11.
With `-T` is truthy by default, and `-t` is falsy by default.

**Review by carlosala — CHANGES_REQUESTED** [review]
I think it'd be better to keep arguments separated, instead of trying to join `$2` with the head. For example, if `head` and `$2` are empty, we end up with `eza - <tail>`

**Comment by pepoluan** [comment]
> I think it'd be better to keep arguments separated, instead of trying to join `$2` with the head. For example, if `head` and `$2` are empty, we end up with `eza - <tail>`

If it's within a function then I agree. But I'd like to keep it compact because it will become an alias, and for me it's a bit cluttery if the head (single-char options) part are separated.

But you do highlight an edge case that I had overlooked; my next commit will fix this.

File: plugins/eza/eza.plugin.zsh
Comment (pepoluan) [reply plugins/eza/eza.plugin.zsh:25]: Ahh good idea! Never knew about the `-T`|`-t` flag before!

Comment (pepoluan) [reply plugins/eza/eza.plugin.zsh]: Implemented in commit [c7c7dd8](https://github.com/ohmyzsh/ohmyzsh/pull/12237/commits/c7c7dd81ac8cb7366c34946d980d104474adc397)

**Review by pepoluan — COMMENTED** [review]
All suggested changes have been considered and adapted.

File: plugins/eza/README.md
Comment (carlosala) [inline plugins/eza/README.md:1]: Could you use `yes/no` instead of `true/false`? That's the design that we are using across the project!

**Review by carlosala — CHANGES_REQUESTED** [review]

**Comment by carlosala** [comment]
Do you want to be the CODEOWNER for the plugin? Then you'll be notified every time someone modifies it! Feel free to modify https://github.com/ohmyzsh/ohmyzsh/blob/master/.github/CODEOWNERS and add you!

**Comment by pepoluan** [comment]
> Do you want to be the CODEOWNER for the plugin? Then you'll be notified every time someone modifies it! Feel free to modify https://github.com/ohmyzsh/ohmyzsh/blob/master/.github/CODEOWNERS and add yourself!

Ah okay! Going to add myself there. Thanks!

File: plugins/eza/README.md
Comment (pepoluan) [reply plugins/eza/README.md:1]: Ah okay, I can change the doc. Using `yes/no` will also be picked up by `-T/-t` right?

Comment (pepoluan) [reply plugins/eza/README.md:1]: I will miss `vim` highlighting the `true|false` value though...

image
image

But I'll defer to the current convention.

Implemented in commit [a8e7178](https://github.com/ohmyzsh/ohmyzsh/pull/12237/commits/a8e7178fe7411a7f31bfb7aecf0f5f6da6b19327)

**Review by carlosala — APPROVED** [review]
I changed the `showgroup` to `show-group` for consistency! LGTM! 🚀

**Pull request merged**
